### PR TITLE
Upload the regression file to Amazon S3

### PR DESCRIPTION
**Pull request by xuzhao9**
When a regression happens, we should also upload the regression detector output to S3.

Test plan:

Local testing:
```
$ python ./scripts/userbenchmark/upload_s3.py --userbenchmark_json $LATEST_REGRESSION_RESULT --userbenchmark_platform "gcp_a100"
```

### CODE REVIEW

File: scripts/userbenchmark/upload_s3.py [hunk 1]
```diff
@@ -34,7 +33,7 @@ def upload_s3(ub_name: str, platform_name: str, date_str: str, file_path: Path):
     parser.add_argument("--userbenchmark_platform", required=True,
                         help='Name of the userbenchmark platform')
     parser.add_argument("--userbenchmark_json", required=True,
-                        help='Upload userbenchmark json data')
+                        help='Upload userbenchmark json or regression yaml file.')
```
Comment (janeyx99) [inline scripts/userbenchmark/upload_s3.py:36]:
```suggestion
                        help='Upload userbenchmark json and regression yaml file.')
```
I forget what this flag does now...is it just a True/False whether we upload stats?

**Review by janeyx99 — APPROVED** [review]
Seems fine though I don't know if it'd be worth removing the slight duplication in get_date_from_metrics.

File: scripts/userbenchmark/upload_s3.py (hunk 1 above)
Comment (xuzhao9) [reply scripts/userbenchmark/upload_s3.py:36]: It specifies the file we want to upload to S3. Previously it should be just metrics-*.json file, now we want it to upload regression-*.yaml files as well.

Comment (janeyx99) [reply scripts/userbenchmark/upload_s3.py:36]: Hmmmm would this be clearer as just "--file-to-upload" or "--files-to-upload" then?

File: userbenchmark/utils.py [hunk 2]
```diff
@@ -68,15 +69,26 @@ def dump_output(bm_name, output, target_dir: Path=None) -> None:
...
-    return datetime.strftime(datetime_obj, "%Y-%m-%d")
-
+    if metrics_file.startswith("metrics-"):
+        datetime_obj = datetime.strptime(metrics_file, "metrics-%Y%m%d%H%M%S")
+        return datetime.strftime(datetime_obj, "%Y-%m-%d")
+    elif metrics_file.startswith("regression-"):
+        datetime_obj = datetime.strptime(metrics_file, "regression-%Y%m%d%H%M%S")
+        return datetime.strftime(datetime_obj, "%Y-%m-%d")
+    print(f"Unknown metrics or regression file name format: {metrics_file}")
+    exit(1)
```
Comment (janeyx99) [inline userbenchmark/utils.py:79]: Right, but we could refactor like:
```suggestion
    assert metrics_file.startswith("metrics-") or metrics_file.startswith("regression-"), f"Unknown metrics or regression file name format: {metrics_file}"
    # metrics_file usually looks like metrics-%Y%m%d%H%M%S or regression-%Y%m%d%H%M%S
    stripped_filename = metrics_file.split("-")[1]
    datetime_obj = datetime.strptime(metrics_file.split("-")[1], "%Y%m%d%H%M%S")
    return datetime.strftime(datetime_obj, "%Y-%m-%d")
```

Comment (xuzhao9) [reply userbenchmark/utils.py]: Right, this would be cleaner. Thanks!

File: scripts/userbenchmark/upload_s3.py (hunk 1 above)
Comment (xuzhao9) [reply scripts/userbenchmark/upload_s3.py:36]: Right, let me change it to "--upload-file".

**Pull request closed without merging**